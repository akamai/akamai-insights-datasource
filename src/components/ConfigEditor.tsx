import React, { ChangeEvent } from 'react';
import { InlineField, Input } from '@grafana/ui';
import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
import { MyDataSourceOptions } from '../types';

interface Props extends DataSourcePluginOptionsEditorProps<MyDataSourceOptions> {}

export function ConfigEditor(props: Props) {
  const { options } = props;

  const onClientSecretChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { onOptionsChange, options } = props;
    const jsonData = {
      ...options.jsonData,
      clientSecret: event.target.value,
    };
    onOptionsChange({ ...options, jsonData });
  };

  const onHostChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { onOptionsChange, options } = props;
    const jsonData = {
      ...options.jsonData,
      host: event.target.value,
    };
    onOptionsChange({ ...options, jsonData });
  };

  const onAccessTokenChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { onOptionsChange, options } = props;
    const jsonData = {
      ...options.jsonData,
      accessToken: event.target.value,
    };
    // console.log('host: ' + event.target.value);
    onOptionsChange({ ...options, jsonData });
  };

  const onClientTokenChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { onOptionsChange, options } = props;
    const jsonData = {
      ...options.jsonData,
      clientToken: event.target.value,
    };
    // console.log('host: ' + event.target.value);
    onOptionsChange({ ...options, jsonData });
  };


  const { jsonData } = options;


  return (
    <div className="gf-form-group">
      <InlineField  label="Akamai Client Secret" labelWidth={20}>
        <Input
          onChange={onClientSecretChange}
          value={jsonData.clientSecret || ''}
          placeholder="Enter client secret"
          width={40}
        />
      </InlineField>
      <InlineField label="Host" labelWidth={20}>
        <Input
          value={jsonData.host || ''}
          placeholder="Enter host"
          width={40}
          onChange={onHostChange}
        />
      </InlineField>
      <InlineField label="Access Token" labelWidth={20}>
        <Input
            value={jsonData.accessToken || ''}
            placeholder="Enter access token"
            width={40}
            onChange={onAccessTokenChange}
        />
      </InlineField>
      <InlineField label="Client Token" labelWidth={20}>
        <Input
            value={jsonData.clientToken || ''}
            placeholder="Enter client token"
            width={40}
            onChange={onClientTokenChange}
        />
      </InlineField>
    </div>
  );
}
