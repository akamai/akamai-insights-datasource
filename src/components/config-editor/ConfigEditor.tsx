import { InlineField, Input } from '@grafana/ui';
import React, { ChangeEvent } from 'react';

import { DataSourceProps } from '../../types/types';
import { Secret } from './types';

export function ConfigEditor({ options, onOptionsChange }: DataSourceProps) {

  const onInputChange = ({ target: { value } }: ChangeEvent<HTMLInputElement>, fieldName: string) => {
    const jsonData = {
      ...options.jsonData,
      [ fieldName ]: value
    };
    onOptionsChange({ ...options, jsonData });
  };

  const onClientSecretChange = (event: ChangeEvent<HTMLInputElement>) => onInputChange(event, Secret.ClientSecret);
  const onHostChange = (event: ChangeEvent<HTMLInputElement>) => onInputChange(event, Secret.Host);
  const onAccessTokenChange = (event: ChangeEvent<HTMLInputElement>) => onInputChange(event, Secret.AccessToken);
  const onClientTokenChange = (event: ChangeEvent<HTMLInputElement>) => onInputChange(event, Secret.ClientToken);

  const { jsonData: { clientSecret, host, clientToken, accessToken } } = options;

  return (
    <div className="gf-form-group">
      <InlineField
        label="Akamai Client Secret"
        labelWidth={20}>
        <Input
          onChange={onClientSecretChange}
          value={clientSecret || ''}
          placeholder="Enter client secret"
          width={40}
        />
      </InlineField>
      <InlineField
        label="Host"
        labelWidth={20}>
        <Input
          value={host || ''}
          placeholder="Enter host"
          width={40}
          onChange={onHostChange}
        />
      </InlineField>
      <InlineField
        label="Access Token"
        labelWidth={20}>
        <Input
          value={accessToken || ''}
          placeholder="Enter access token"
          width={40}
          onChange={onAccessTokenChange}
        />
      </InlineField>
      <InlineField
        label="Client Token"
        labelWidth={20}>
        <Input
          value={clientToken || ''}
          placeholder="Enter client token"
          width={40}
          onChange={onClientTokenChange}
        />
      </InlineField>
    </div>
  );
}
