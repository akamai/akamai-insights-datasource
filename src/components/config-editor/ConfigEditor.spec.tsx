import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import { ConfigEditor } from './ConfigEditor';

describe('ConfigEditor component', () => {
  it('should trigger onOptionsChange when one of inputs has changed', () => {
    const options: any = {
      jsonData: {
        clientToken: 'testToken'
      }
    };
    const onOptionsChange = jest.fn();

    const { getByPlaceholderText } = render(
      <ConfigEditor options={options} onOptionsChange={onOptionsChange}/>
    );

    fireEvent.change(getByPlaceholderText('Enter host'), { target: { value: 'akamai.com' } });

    expect(onOptionsChange).toHaveBeenCalledWith({
      jsonData: {
        clientToken: 'testToken',
        host: 'akamai.com'
      }
    });
  });
});
