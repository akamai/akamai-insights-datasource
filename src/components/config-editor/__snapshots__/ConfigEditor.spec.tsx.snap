// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigEditor component should mark textarea as invalid and not update form when credentials are in wrong format 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="gf-form-group"
      >
        <div
          class="credentials__container"
        >
          <div
            class="css-1rplq84"
          >
            <div
              class="css-l4ykjo-Label"
            >
              <label>
                <div
                  class="css-70qvj9"
                >
                  Information
                </div>
                <span
                  class="css-1qya104-Label-description"
                >
                  Paste credentials here to auto fill form below
                </span>
              </label>
            </div>
            <div>
              <div>
                <textarea
                  class="css-i28uzi credentials__textarea"
                  name="credentialsTextArea"
                />
              </div>
            </div>
          </div>
        </div>
        <hr />
        <div
          class="css-1uqla6q"
        >
          <label
            class="css-1hzpsg9"
          >
            Akamai Client Secret
          </label>
          <div
            class="css-1tu59u4"
          >
            <div
              class="css-1xvl4yl-input-wrapper"
              data-testid="input-wrapper"
            >
              <div
                class="css-10lnb82-input-inputWrapper"
              >
                <input
                  class="css-8tk2dk-input-input"
                  placeholder="Enter client secret"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1uqla6q"
        >
          <label
            class="css-1hzpsg9"
          >
            Host
          </label>
          <div
            class="css-1tu59u4"
          >
            <div
              class="css-1xvl4yl-input-wrapper"
              data-testid="input-wrapper"
            >
              <div
                class="css-10lnb82-input-inputWrapper"
              >
                <input
                  class="css-8tk2dk-input-input"
                  placeholder="Enter host"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1uqla6q"
        >
          <label
            class="css-1hzpsg9"
          >
            Access Token
          </label>
          <div
            class="css-1tu59u4"
          >
            <div
              class="css-1xvl4yl-input-wrapper"
              data-testid="input-wrapper"
            >
              <div
                class="css-10lnb82-input-inputWrapper"
              >
                <input
                  class="css-8tk2dk-input-input"
                  placeholder="Enter access token"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1uqla6q"
        >
          <label
            class="css-1hzpsg9"
          >
            Client Token
          </label>
          <div
            class="css-1tu59u4"
          >
            <div
              class="css-1xvl4yl-input-wrapper"
              data-testid="input-wrapper"
            >
              <div
                class="css-10lnb82-input-inputWrapper"
              >
                <input
                  class="css-8tk2dk-input-input"
                  placeholder="Enter client token"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="gf-form-group"
    >
      <div
        class="credentials__container"
      >
        <div
          class="css-1rplq84"
        >
          <div
            class="css-l4ykjo-Label"
          >
            <label>
              <div
                class="css-70qvj9"
              >
                Information
              </div>
              <span
                class="css-1qya104-Label-description"
              >
                Paste credentials here to auto fill form below
              </span>
            </label>
          </div>
          <div>
            <div>
              <textarea
                class="css-i28uzi credentials__textarea"
                name="credentialsTextArea"
              />
            </div>
          </div>
        </div>
      </div>
      <hr />
      <div
        class="css-1uqla6q"
      >
        <label
          class="css-1hzpsg9"
        >
          Akamai Client Secret
        </label>
        <div
          class="css-1tu59u4"
        >
          <div
            class="css-1xvl4yl-input-wrapper"
            data-testid="input-wrapper"
          >
            <div
              class="css-10lnb82-input-inputWrapper"
            >
              <input
                class="css-8tk2dk-input-input"
                placeholder="Enter client secret"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1uqla6q"
      >
        <label
          class="css-1hzpsg9"
        >
          Host
        </label>
        <div
          class="css-1tu59u4"
        >
          <div
            class="css-1xvl4yl-input-wrapper"
            data-testid="input-wrapper"
          >
            <div
              class="css-10lnb82-input-inputWrapper"
            >
              <input
                class="css-8tk2dk-input-input"
                placeholder="Enter host"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1uqla6q"
      >
        <label
          class="css-1hzpsg9"
        >
          Access Token
        </label>
        <div
          class="css-1tu59u4"
        >
          <div
            class="css-1xvl4yl-input-wrapper"
            data-testid="input-wrapper"
          >
            <div
              class="css-10lnb82-input-inputWrapper"
            >
              <input
                class="css-8tk2dk-input-input"
                placeholder="Enter access token"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1uqla6q"
      >
        <label
          class="css-1hzpsg9"
        >
          Client Token
        </label>
        <div
          class="css-1tu59u4"
        >
          <div
            class="css-1xvl4yl-input-wrapper"
            data-testid="input-wrapper"
          >
            <div
              class="css-10lnb82-input-inputWrapper"
            >
              <input
                class="css-8tk2dk-input-input"
                placeholder="Enter client token"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
